lib/dockly/docker.rb

```diff
@@ -33,12 +33,27 @@ def build_env(hash = nil)
   end
 
   def generate!
-    image = generate_build
+    image = find_image_by_repotag || generate_build
     export_image(image)
   ensure
     cleanup([image]) if cleanup_images
   end
 
+  def export_only
+    if image = find_image_by_repotag
+      info "Found image by repo:tag: #{repo}:#{tag} - #{image.inspect}"
+      export_image(image)
+    else
+      raise "Could not find image"
+    end
+  end
+
+  def find_image_by_repotag
```

Comment:
True. I just didn't want someone to run `dockly:docker:prepare:x` and then validate, then run `dockly:docker:upload:x` and have it rebuild the whole image.